dkc/core/models/user_quota.py

```diff
@@ -1,26 +1,29 @@
 from typing import Type
 
+from django.apps import apps
 from django.contrib.auth.models import User
 from django.db import models
 from django.db.models.signals import post_save
 from django.dispatch import receiver
 from django_girders.models import SelectRelatedManager
 
 
-class Quota(models.Model):
-    allocation = models.BigIntegerField(default=0)
-    user = models.OneToOneField(User, null=True, on_delete=models.CASCADE)
-    # root_set
+class UserQuota(models.Model):
+    user = models.OneToOneField(
+        User, primary_key=True, on_delete=models.CASCADE, related_name='quota'
+    )
+    used = models.PositiveBigIntegerField(default=0)
+    allowed = models.PositiveBigIntegerField()
 
     # The user is needed to stringify, so join it when directly querying for a quota
     objects = SelectRelatedManager('user')
 
     def __str__(self):
-        return f'Quota for <{self.user}>; {self.allocation} bytes'
```

Comment:
I'm worried this might actually be a little long. Have you seen it rendered somewhere in the admin page yet?

Reply:
It doesn't currently appear on the admin page. I may just strip some of these things out for now and let us deal with the admin page later. I'll probably also remove the SelectRelatedManager as I don't even think we'll query the quota tables directly, but rather always through FK lookups.